internal/vault/auth_methods.go

```diff
@@ -0,0 +1,475 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vault
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"emperror.dev/errors"
+	"github.com/cristalhq/jwt/v3"
+	"github.com/hashicorp/vault/api"
+	json "github.com/json-iterator/go"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cast"
+)
+
+type auth struct {
+	Type             string                 `json:"type"`
+	Path             string                 `json:"path"`
+	Description      string                 `json:"description"`
+	UsersOrGroupsKey string                 `json:"usersOrGroupsKey"`
+	Roles            []interface{}          `json:"roles"`
+	Users            interface{}            `json:"users"`
+	Crossaccountrole []interface{}          `json:"crossaccountrole"`
+	Groups           map[string]interface{} `json:"groups"`
+	Options          map[string]interface{} `json:"options"`
+	Map              map[string]interface{} `json:"map"`
+	Config           map[string]interface{} `json:"config"`
+}
+
+// getExistingAuthMethods gets all auth methods that are already in Vault.
+// The existing auth methods are in a map to make it easy to disable easily from it with "O(n)" complexity.
+func (v *vault) getExistingAuthMethods() (map[string]*api.MountOutput, error) {
+	existingAuths := make(map[string]*api.MountOutput)
+
+	existingAuthList, err := v.cl.Sys().ListAuth()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to list existing auth methods")
+	}
+
+	for _, existingAuthMethod := range existingAuthList {
+		existingAuths[existingAuthMethod.Type] = existingAuthMethod
+	}
+
+	return existingAuths, nil
+}
+
+// getUnmanagedAuthMethods gets unmanaged auth methods by comparing what's already in Vault and what's in the externalConfig
+func (v *vault) getUnmanagedAuthMethods(managedAuthMethods []auth) map[string]*api.MountOutput {
+	// TODO: remove this line and just use passed value
+	unmangedAuths, _ := v.getExistingAuthMethods()
+	// Remove managed auth methods form the items since the rest will be disabled.
+	for _, mangedAuthMethod := range managedAuthMethods {
+		delete(unmangedAuths, mangedAuthMethod.Type)
+	}
+	// Remove token auth method since it's the default
+	delete(unmangedAuths, "token")
+
+	return unmangedAuths
+}
+
+func (v *vault) configureAuthMethods() error {
+	managedAuths := extConfig.Auth
+	existingAuths, _ := v.getExistingAuthMethods()
+	unManagedAuths := v.getUnmanagedAuthMethods(managedAuths)
```

Comment:
`unmanaged` is a single word so the `M` should be lower case (`unmanagedAuths`).